### PR TITLE
Add linting for tests and code samples

### LINKED ISSUE #635: Configure eslint for tests
We currently don't lint source code of tests which is not nice. The configuration should be placed in the `./test` subdir and `lint` script should be updated in `package.json`.

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Closes #635

Each source code directory (src, test, code_samples) now has its own eslint configuration.
    
Also includes the following:
* Fixes issues in tests and code samples reported by eslint
* Updates readme for code samples

### CODE REVIEW

**Comment by puzpuzpuz** [comment]
@mdumandag I've rebased this one, so it's ready for a review.

File: code_samples/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+    "env": {
+        "mocha": true,
```
Comment (mdumandag) [inline code_samples/.eslintrc:3]: Why do we need `mocha` in code samples?

File: code_samples/org-website/GlobalSerializerSample.js [hunk 2]
```diff
@@ -22,12 +22,12 @@ class GlobalSerializer {
         this.id = 20;
     }
 
-    read(input) {
+    read(input) { // eslint-disable-line no-unused-vars
```
Comment (mdumandag) [inline code_samples/org-website/GlobalSerializerSample.js:25]: Assuming this code sample be visible in the `.org` site, should we display these ignore comments? I know that it makes not much sense but maybe we can use JSON.parse/stringify here to demonstrate the concept?

File: code_samples/.eslintrc (hunk 1 above)
Comment (puzpuzpuz) [reply code_samples/.eslintrc]: Good catch. Addressed in 13cc25c0903a7d4d671c49bd1751eac20f1b161e

File: code_samples/org-website/GlobalSerializerSample.js (hunk 2 above)
Comment (puzpuzpuz) [reply code_samples/org-website/GlobalSerializerSample.js]: Good suggestion. Change the sample in 13cc25c0903a7d4d671c49bd1751eac20f1b161e

**Review by mdumandag — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
@mdumandag thank you so much for the code review!

**Pull request merged**